src/card_operations/card_fetch_wallet_list.c

```diff
@@ -0,0 +1,156 @@
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+#include "card_fetch_wallet_list.h"
+
+#include "card_internal.h"
+#include "card_return_codes.h"
+#include "card_utils.h"
+#include "flash_api.h"
+#include "nfc.h"
+#include "ui_screens.h"
+
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL FUNCTIONS
+ *****************************************************************************/
+bool card_fetch_wallet_list(card_fetch_wallet_list_config_t *configuration,
+                            card_fetch_wallet_list_response_t *response) {
+  bool result = false;
+  if (NULL == configuration || NULL == response ||
+      NULL == response->wallet_list || NULL == response->length) {
+    return result;
+  }
+
+  // Render the instruction screen
+  instruction_scr_init(configuration->msg, configuration->heading);
+
+  card_operation_data_t card_data = {0};
+  card_data.nfc_data.retries = 5;
+  card_data.nfc_data.init_session_keys = true;
+
+  while (1) {
+    memcpy(card_data.nfc_data.family_id, get_family_id(), FAMILY_ID_SIZE);
+    card_data.nfc_data.acceptable_cards = configuration->acceptable_cards;
+
+    card_initialize_applet(&card_data);
+    response->tapped_card = card_data.nfc_data.tapped_card;
+
+    if (CARD_OPERATION_SUCCESS == card_data.error_type) {
+      card_data.nfc_data.status =
+          nfc_list_all_wallet(response->wallet_list, response->length);
+      // Last 2 bytes of the APDU contain the SW
+      *response->length -= 2;
```

Comment:
Add a TODO: either here or in the API to update this handling inside `nfc_list_all_wallet` itself.

Reply:
Added a TODO